src/test.js

```diff
@@ -733,10 +733,16 @@ function internalStart( test ) {
 }
 
 function numberOfTests( module ) {
-	var count = module.tests.length;
-	while ( ( module = module.childModule ) ) {
-		count += module.tests.length;
+	let count = module.tests.length,
+		modules = [ ...module.childModules ];
```

Comment:
`Array.from` would be a better option here as it shows better that you want to clone this array. Even though, it's a built-in api I suspect it's not present in every supported browser. 


---

`var` is not deprecated or even close to this. `let` is made for block scoped bindings. `const` too, but it also goes forward protecting a name to change it's value. 

I don't see a reason for us to be super protective with our bindings and `const` everything that are not assigned more than once. This adds logic (to prevent reassignments) we particularly don't need in most cases.

In any way, the best answer should be a convention. @jzaefferer is there any discussion about it for the jQuery Code Style?